axlearn/common/attention.py

```diff
@@ -1812,6 +1813,10 @@ def _forward_for_mode(
             query_positions = query_positions + time_step[:, None]  # [batch, steps]
         q_proj, k_proj, v_proj = self.i_proj(query, query_positions=query_positions, **kv_kwargs)
 
```

Comment:
don't add here, add these in envy.py file based on whether we are using CP or not.